src/pages/iou/steps/IOUAmountPage/index.js

```diff
@@ -0,0 +1,179 @@
+        // Cleanup all keydown event listeners that we've set up
+        if (document) {
+            document.removeEventListener('keydown', this.keyDownHandler);
+        }
+    }
+
+    /**
+     * Update amount with number or Backspace pressed.
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit to enable Next button
+     *
+     * @param {String} key
+     */
+    updateAmountIfValidInput(key) {
+        // Backspace button is pressed
+        if (key === '<' || key === 'Backspace') {
+            if (this.state.amount.length > 0) {
+                this.setState(prevState => ({
+                    amount: prevState.amount.substring(0, prevState.amount.length - 1),
+                }));
+            }
+            return;
+        }
+
+        this.setState((prevState) => {
+            const newValue = `${prevState.amount}${key}`;
+
+            // Regex to validate decimal number with up to 3 decimal numbers
+            const decimalNumberRegex = new RegExp(/^\d+(\.\d{0,3})?$/, 'i');
+            if (!decimalNumberRegex.test(newValue)) {
+                return prevState;
+            }
+            return {
+                amount: newValue,
+            };
+        });
+    }
+
+    render() {
+        return (
+            <View style={[styles.flex1, styles.pageWrapper]}>
+                <View style={[
+                    styles.flex1,
+                    styles.flexRow,
+                    styles.w100,
+                    styles.alignItemsCenter,
+                    styles.justifyContentCenter,
+                ]}
+                >
+                    <Text style={styles.iouAmountText}>
+                        {this.props.selectedCurrency}
+                    </Text>
+                    {this.props.isSmallScreenWidth
+                        ? <Text style={styles.iouAmountText}>{this.state.amount}</Text>
+                        : (
+                            <TextInputAutoWidth
+                                    inputStyle={styles.iouAmountTextInput}
+                                    textStyle={styles.iouAmountText}
+                                    onKeyPress={(event) => {
+                                        this.updateAmountIfValidInput(event.key);
+                                        event.preventDefault();
+                                    }}
+                                    ref={el => this.textInput = el}
+                                    value={this.state.amount}
+                            />
+                        )}
+                </View>
+                <View style={[styles.w100, styles.justifyContentEnd]}>
+                    {this.props.isSmallScreenWidth
+                        ? <BigNumberPad numberPressed={this.updateAmountIfValidInput} />
+                        : <View />}
+                    <TouchableOpacity
+                            style={[styles.button, styles.w100, styles.mt5, styles.buttonSuccess,
```

Comment:
Indentation is off here too.

Reply:
Ran into that weird eslint bug but did the workaround